code/jumpnode/jumpnode.cpp

```diff
@@ -392,6 +405,25 @@ CJumpNode *jumpnode_get_by_name(const char* name)
 	return NULL;
 }
 
+/**
+ * Get jump node object by the object number
+ *
+ * @param objnum to search for
+ * @return Jump node object pointer
+ */
+CJumpNode *jumpnode_get_by_objnum(int objnum)
+{
+	Assert(objnum > -1);
+	SCP_list<CJumpNode>::iterator jnp;
+
+	for (jnp = Jump_nodes.begin(); jnp != Jump_nodes.end(); ++jnp) {
```

Comment:
You should be able to use a for-each loop here.

Reply:
Do I use `Jump_nodes.size()` to indicate the range?

Reply:
You should be able to do something like this:
```c++
for (const CJumpNode& jnp : Jump_nodes)
```

Reply:
It didn't like const, but it should be this type now. I should compile and test one more time to make sure it works.

Reply:
I was able to test, and it still works.  👍